test.py

```diff
@@ -0,0 +1,467 @@
+#!/usr/bin/env python3
+import os
+import sys
+
+import argparse
+import logging
+import shutil
+import subprocess
+import tempfile
+import time
+
+
+def get_repo_root():
+    return os.path.dirname(sys.argv[0])
+
+
+logger = logging.getLogger("dcos-commons-test")
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(name)s %(message)s")
+
+sys.path.append(os.path.join(get_repo_root(), 'tools'))
+import clustinfo
+import fwinfo
+import launch_ccm_cluster
+
+work_dir = None
+def get_work_dir():
+    global work_dir
+    if not work_dir:
+        work_dir = tempfile.mkdtemp(prefix='test_workdir', dir=get_repo_root())
+        logger.info("Using %s for test run files", work_dir)
+    return work_dir
+
+def parse_args(args=sys.argv):
+    parser = argparse.ArgumentParser(description="Optionally build and test dcos-commons frameworks")
+    parser.add_argument("--test-only", action='store_false', dest='run_build',
+            help="requires a prior build")
+    parser.add_argument("--build-only", action='store_false', dest='run_tests')
+    parser.add_argument("--parallel", action='store_true',
+            help="Use more than one cluster to try to speed up the tests")
+    parser.add_argument("--cluster-count", type=int, default=2,
+            help="Ignored unless using --parallel.  "
+                 "Number of clusters to run tests on. "
+                 "Large values are likely to break CCM.")
+    parser.add_argument("--order", choices=('random', 'ordered'), default='random',
+            help="Run tests in randomr order, or the order given on the command line.  "
```

Comment:
the ordering isn't implemented here, the full args were more or less trying to plan what I needed to support, and the pull request was kind of a minimum "this runs at all, and please stop me if this is awful"